x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -6,59 +6,37 @@
 
 import theme from '@elastic/eui/dist/eui_theme_light.json';
 import {
+  first,
   flatten,
   groupBy,
-  indexBy,
+  isEmpty,
   sortBy,
   uniq,
-  zipObject,
-  isEmpty,
-  first
+  zipObject
 } from 'lodash';
 import { TraceAPIResponse } from '../../../../../../../../server/lib/traces/get_trace';
 import { Span } from '../../../../../../../../typings/es_schemas/ui/Span';
 import { Transaction } from '../../../../../../../../typings/es_schemas/ui/Transaction';
 
-interface IWaterfallIndex {
-  [key: string]: IWaterfallItem | undefined;
-}
-
 interface IWaterfallGroup {
   [key: string]: IWaterfallItem[];
 }
 
 export interface IWaterfall {
-  entryTransaction?: Transaction;
-  traceRoot?: Transaction;
-  traceRootDuration?: number;
-
   /**
    * Duration in us
    */
   duration: number;
-  services: string[];
-  orderedItems: IWaterfallItem[];
-  itemsById: IWaterfallIndex;
-  getTransactionById: (id?: IWaterfallItem['id']) => Transaction | undefined;
-  errorCountByTransactionId: TraceAPIResponse['errorsPerTransaction'];
   serviceColors: IServiceColors;
+  entryTransaction?: Transaction;
+  rootTransaction?: Transaction;
+  errorsPerTransaction: TraceAPIResponse['errorsPerTransaction'];
+  items: IWaterfallItem[];
 }
 
 interface IWaterfallItemBase {
-  id: string | number;
-  parentId?: string;
-  serviceName: string;
-  name: string;
-
-  /**
-   * Duration in us
-   */
-  duration: number;
-
-  /**
-   * start timestamp in us
-   */
-  timestamp: number;
+  parent?: IWaterfallItem;
```

Comment:
I added the parent here to remove the `getTransactionById` from the returned object.